zim/tokenparser.py

```diff
@@ -32,12 +39,26 @@ def collect_untill_end_token(token_iter, tag):
 			nesting -= 1
 			if nesting < 0:
 				break
-
-		tokens.append(t)
+		yield t
 	else:
 		raise EndOfTokenListError('Did not find "%s" closing tag' % tag)
 
-	return tokens
+
+def filter_token(token_iter, token):
```

Comment:
Good addition to the toolkit. Please leave this function in, even if the patch doesn't need it anymore (due to fix in 5b87aa73e3c96772c81dab692a33ef1acbc2c42e).

Thanks!